client/src/components/SummaryCalculation.ts

```diff
@@ -0,0 +1,83 @@
+import _ from "lodash";
+import helpers from "util/helpers";
+import { AddressLocation, AddressRecord, SummaryStatsRecord } from "./APIDataTypes";
+
+export const getTopFiveContactsInPortfolio = (addrs: AddressRecord[]) => {
+  const allContactNames = addrs
+    .map((a) => a.ownernames?.map((ownername) => ownername.value) || [])
+    .flat();
+
+  return helpers.getMostCommonElementsInArray(allContactNames, 5);
+};
+
+export const extractLocationDataFromAddr = (addr: AddressRecord): AddressLocation => {
+  return (({ boro, housenumber, lat, lng, streetname }) => ({
+    boro,
+    housenumber,
+    lat,
+    lng,
+    streetname,
+  }))({
+    ...addr,
+  });
+};
+
+export const calculateAggDataFromAddressList = (addrs: AddressRecord[]): SummaryStatsRecord => {
+  const bldgs = addrs.length;
+  const units = _.sumBy(addrs, (a) => a.unitsres || 0);
+
+  const totalopenviolations = _.sumBy(addrs, (a) => a.openviolations);
```

Comment:
should these be in camelCase?

Reply:
hmmm so they normally would but since our API responses give their fields in lower case like this (bc I think the SQL query result has them like this), I think it would be good to keep things consistent with the other big objects of data we house in APIDataTypes. Does that make sense?